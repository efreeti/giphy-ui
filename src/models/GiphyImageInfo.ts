export interface GiphyImageInfo {
  url: string,
  width: number,
  height: number,
}
