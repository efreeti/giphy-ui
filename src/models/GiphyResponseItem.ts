import { GiphyImageInfo } from "@/models/GiphyImageInfo";

export interface GiphyResponseItem {
  id: string,
  images: {
    downsized_large: GiphyImageInfo,
  },
}
