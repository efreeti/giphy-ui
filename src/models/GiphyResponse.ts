import { GiphyResponseItem } from "@/models/GiphyResponseItem";

export interface GiphyResponse {
  data: Array<GiphyResponseItem>,
  pagination: {
    total_count: number,
  },
}
