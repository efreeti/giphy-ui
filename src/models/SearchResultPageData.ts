import { GiphyResponseItem } from "@/models/GiphyResponseItem";

export class SearchResultPageData {
  constructor(
    public loaded: boolean,
    public images: Array<GiphyResponseItem>,
    public error: any,
  ) {}
}
