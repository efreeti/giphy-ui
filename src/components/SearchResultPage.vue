<template>
  <div>
    <div v-for="image in pageData.images" v-bind:key="image.id">
      <image-card>
        <giphy-image :image-info="image.images.downsized_large" />
      </image-card>
    </div>

    <div v-if="!pageData.loaded || pageData.error">
      <div v-for="n in pageMaxSize" v-bind:key="n">
        <image-card>
          <image-placeholder>
            <error-message v-if="pageData.error">
              Something went wrong, try refreshing the page
            </error-message>
          </image-placeholder>
        </image-card>
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

import { SearchResultPageData } from '@/models/SearchResultPageData';

import GiphyImage from "@/components/GiphyImage.vue";
import ImagePlaceholder from "@/components/ImagePlaceholder.vue";
import ImageCard from "@/components/ImageCard.vue";
import ErrorMessage from "@/components/ErrorMessage.vue";

@Component({
  components: {
    'error-message': ErrorMessage,
    'giphy-image': GiphyImage,
    'image-placeholder': ImagePlaceholder,
    'image-card': ImageCard,
  }
})
export default class SearchResultPageView extends Vue {
  @Prop() public pageData!: SearchResultPageData;

  @Prop() public pageMaxSize!: number;
}
</script>
