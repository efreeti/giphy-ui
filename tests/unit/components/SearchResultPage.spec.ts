import { mount } from '@vue/test-utils';
import { intersectionObserver } from '@shopify/jest-dom-mocks';

import '../config';

import { SearchResultPageData } from '@/models/SearchResultPageData';
import SearchResultPage from '@/components/SearchResultPage.vue';
import { createImages } from '../utils';

describe('SearchResultPage', () => {
  beforeEach(() => {
    intersectionObserver.mock();
  });

  afterEach(() => {
    intersectionObserver.restore();
  });

  it('should render loading state correctly', () => {
    expect(mount(SearchResultPage, {
      propsData: {
        pageData: new SearchResultPageData(false, [], null),
        pageMaxSize: 12,
      },
    }))
        .toMatchSnapshot();
  });

  it('should render success state correctly', () => {
    expect(mount(SearchResultPage, {
      propsData: {
        pageData: new SearchResultPageData(
          true, createImages([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]), null
        ),
        pageMaxSize: 12,
      },
    }))
        .toMatchSnapshot();
  });

  it('should render error state correctly', () => {
    expect(mount(SearchResultPage, {
      propsData: {
        pageData: new SearchResultPageData(true, [], new Error('Some error')),
        pageMaxSize: 12,
      },
    }))
        .toMatchSnapshot();
  });
});
