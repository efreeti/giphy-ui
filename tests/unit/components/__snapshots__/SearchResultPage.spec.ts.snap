// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultPage should render error state correctly 1`] = `
<div>
  <div>
    <div>
      <div class="image-card"><span class="image-background"><div class="image-placeholder"><span class="image-placeholder-content"><div><div><i class="md-icon md-icon-font md-size-5x md-accent md-theme-default">error</i></div> <div><span class="error-message-text md-body-2">
            Something went wrong, try refreshing the page
          </span></div>
    </div></span>
  </div></span>
</div>
</div>
<div>
  <div class="image-card"><span class="image-background"><div class="image-placeholder"><span class="image-placeholder-content"><div><div><i class="md-icon md-icon-font md-size-5x md-accent md-theme-default">error</i></div> <div><span class="error-message-text md-body-2">
            Something went wrong, try refreshing the page
          </span></div>
</div></span></div></span></div>
</div>
<div>
  <div class="image-card"><span class="image-background"><div class="image-placeholder"><span class="image-placeholder-content"><div><div><i class="md-icon md-icon-font md-size-5x md-accent md-theme-default">error</i></div> <div><span class="error-message-text md-body-2">
            Something went wrong, try refreshing the page
          </span></div>
</div></span></div></span></div>
</div>
<div>
  <div class="image-card"><span class="image-background"><div class="image-placeholder"><span class="image-placeholder-content"><div><div><i class="md-icon md-icon-font md-size-5x md-accent md-theme-default">error</i></div> <div><span class="error-message-text md-body-2">
            Something went wrong, try refreshing the page
          </span></div>
</div></span></div></span></div>
</div>
<div>
  <div class="image-card"><span class="image-background"><div class="image-placeholder"><span class="image-placeholder-content"><div><div><i class="md-icon md-icon-font md-size-5x md-accent md-theme-default">error</i></div> <div><span class="error-message-text md-body-2">
            Something went wrong, try refreshing the page
          </span></div>
</div></span></div></span></div>
</div>
<div>
  <div class="image-card"><span class="image-background"><div class="image-placeholder"><span class="image-placeholder-content"><div><div><i class="md-icon md-icon-font md-size-5x md-accent md-theme-default">error</i></div> <div><span class="error-message-text md-body-2">
            Something went wrong, try refreshing the page
          </span></div>
</div></span></div></span></div>
</div>
<div>
  <div class="image-card"><span class="image-background"><div class="image-placeholder"><span class="image-placeholder-content"><div><div><i class="md-icon md-icon-font md-size-5x md-accent md-theme-default">error</i></div> <div><span class="error-message-text md-body-2">
            Something went wrong, try refreshing the page
          </span></div>
</div></span></div></span></div>
</div>
<div>
  <div class="image-card"><span class="image-background"><div class="image-placeholder"><span class="image-placeholder-content"><div><div><i class="md-icon md-icon-font md-size-5x md-accent md-theme-default">error</i></div> <div><span class="error-message-text md-body-2">
            Something went wrong, try refreshing the page
          </span></div>
</div></span></div></span></div>
</div>
<div>
  <div class="image-card"><span class="image-background"><div class="image-placeholder"><span class="image-placeholder-content"><div><div><i class="md-icon md-icon-font md-size-5x md-accent md-theme-default">error</i></div> <div><span class="error-message-text md-body-2">
            Something went wrong, try refreshing the page
          </span></div>
</div></span></div></span></div>
</div>
<div>
  <div class="image-card"><span class="image-background"><div class="image-placeholder"><span class="image-placeholder-content"><div><div><i class="md-icon md-icon-font md-size-5x md-accent md-theme-default">error</i></div> <div><span class="error-message-text md-body-2">
            Something went wrong, try refreshing the page
          </span></div>
</div></span></div></span></div>
</div>
<div>
  <div class="image-card"><span class="image-background"><div class="image-placeholder"><span class="image-placeholder-content"><div><div><i class="md-icon md-icon-font md-size-5x md-accent md-theme-default">error</i></div> <div><span class="error-message-text md-body-2">
            Something went wrong, try refreshing the page
          </span></div>
</div></span></div></span></div>
</div>
<div>
  <div class="image-card"><span class="image-background"><div class="image-placeholder"><span class="image-placeholder-content"><div><div><i class="md-icon md-icon-font md-size-5x md-accent md-theme-default">error</i></div> <div><span class="error-message-text md-body-2">
            Something went wrong, try refreshing the page
          </span></div>
</div></span></div></span></div>
</div>
</div>
</div>
`;

exports[`SearchResultPage should render loading state correctly 1`] = `
<div>
  <div>
    <div>
      <div class="image-card"><span class="image-background"><div class="image-placeholder"><span class="image-placeholder-content"></span></div></span>
    </div>
  </div>
  <div>
    <div class="image-card"><span class="image-background"><div class="image-placeholder"><span class="image-placeholder-content"></span></div></span>
  </div>
</div>
<div>
  <div class="image-card"><span class="image-background"><div class="image-placeholder"><span class="image-placeholder-content"></span></div></span>
</div>
</div>
<div>
  <div class="image-card"><span class="image-background"><div class="image-placeholder"><span class="image-placeholder-content"></span></div></span>
</div>
</div>
<div>
  <div class="image-card"><span class="image-background"><div class="image-placeholder"><span class="image-placeholder-content"></span></div></span>
</div>
</div>
<div>
  <div class="image-card"><span class="image-background"><div class="image-placeholder"><span class="image-placeholder-content"></span></div></span>
</div>
</div>
<div>
  <div class="image-card"><span class="image-background"><div class="image-placeholder"><span class="image-placeholder-content"></span></div></span>
</div>
</div>
<div>
  <div class="image-card"><span class="image-background"><div class="image-placeholder"><span class="image-placeholder-content"></span></div></span>
</div>
</div>
<div>
  <div class="image-card"><span class="image-background"><div class="image-placeholder"><span class="image-placeholder-content"></span></div></span>
</div>
</div>
<div>
  <div class="image-card"><span class="image-background"><div class="image-placeholder"><span class="image-placeholder-content"></span></div></span>
</div>
</div>
<div>
  <div class="image-card"><span class="image-background"><div class="image-placeholder"><span class="image-placeholder-content"></span></div></span>
</div>
</div>
<div>
  <div class="image-card"><span class="image-background"><div class="image-placeholder"><span class="image-placeholder-content"></span></div></span>
</div>
</div>
</div>
</div>
`;

exports[`SearchResultPage should render success state correctly 1`] = `
<div>
  <div>
    <div class="image-card"><span class="image-background"><span><div role="placeholder" style="width: 100px; height: 100px;"></div></span></span></div>
  </div>
  <div>
    <div class="image-card"><span class="image-background"><span><div role="placeholder" style="width: 100px; height: 100px;"></div></span></span></div>
  </div>
  <div>
    <div class="image-card"><span class="image-background"><span><div role="placeholder" style="width: 100px; height: 100px;"></div></span></span></div>
  </div>
  <div>
    <div class="image-card"><span class="image-background"><span><div role="placeholder" style="width: 100px; height: 100px;"></div></span></span></div>
  </div>
  <div>
    <div class="image-card"><span class="image-background"><span><div role="placeholder" style="width: 100px; height: 100px;"></div></span></span></div>
  </div>
  <div>
    <div class="image-card"><span class="image-background"><span><div role="placeholder" style="width: 100px; height: 100px;"></div></span></span></div>
  </div>
  <div>
    <div class="image-card"><span class="image-background"><span><div role="placeholder" style="width: 100px; height: 100px;"></div></span></span></div>
  </div>
  <div>
    <div class="image-card"><span class="image-background"><span><div role="placeholder" style="width: 100px; height: 100px;"></div></span></span></div>
  </div>
  <div>
    <div class="image-card"><span class="image-background"><span><div role="placeholder" style="width: 100px; height: 100px;"></div></span></span></div>
  </div>
  <div>
    <div class="image-card"><span class="image-background"><span><div role="placeholder" style="width: 100px; height: 100px;"></div></span></span></div>
  </div>
  <!---->
</div>
`;
